src/shared/libfido2-util.c

```diff
@@ -652,32 +651,27 @@ int fido2_generate_hmac_hash(
                 return log_error_errno(SYNTHETIC_ERRNO(EIO),
                                        "Failed to add FIDO2 assertion credential ID: %s", sym_fido_strerr(r));
 
+        log_info("Generating secret key on FIDO2 security token.");
+
         if (has_up) {
-                r = sym_fido_assert_set_up(a, FIDO_OPT_FALSE);
+                r = sym_fido_assert_set_up(a, FLAGS_SET(lock_with, FIDO2ENROLL_UP) ? FIDO_OPT_TRUE : FIDO_OPT_FALSE);
                 if (r != FIDO_OK)
                         return log_error_errno(SYNTHETIC_ERRNO(EIO),
-                                               "Failed to turn off FIDO2 assertion user presence: %s", sym_fido_strerr(r));
+                                               "Failed to %s FIDO2 assertion user presence: %s",
+                                               enable_disable(FLAGS_SET(lock_with, FIDO2ENROLL_UP)),
+                                               sym_fido_strerr(r));
+
+                if (FLAGS_SET(lock_with, FIDO2ENROLL_UP))
+                        log_notice("%s%sIn order to allow secret key generation, please verify presence on security token.",
```

Comment:
verify → confirm ? I think it's more natural to say "confirm who you are" rather than "verify who you are" when it's just a simple check. Especially that the next patch adds "verification" in a completely different sense.